### ISSUE TITLE
Text is a little blurry

### ISSUE THREAD

**Issue by BenoitCaron** [sm64-port/sm64_3ds#15]
Hi,

I noticed the text is sometimes blurry, especially on the larger font. Small text seems to be fine. Surely this is some scaling issue? Is there a way to fix that?

Regards

![2020-08-05_11-29-29 338_top](https://user-images.githubusercontent.com/5237965/89398159-0bc38400-d711-11ea-9c12-4a88bd577446.png)
![2020-08-05_11-30-15 433_top](https://user-images.githubusercontent.com/5237965/89398170-0fefa180-d711-11ea-9823-68c0d053124a.png)
![2020-08-05_11-30-47 396_top](https://user-images.githubusercontent.com/5237965/89398177-1251fb80-d711-11ea-9e97-9c6683159834.png)

**Comment by Gericom** [comment]
Yeah, I saw that as well. It happens because of the antialiasing. I think the polygons for the text are not properly aligned, making them one-off when rendering at twice the resolution and scaling down. It might be possible to fix that if you can find where that rendering is done (or by simply disabling antialiasing :p)

**Comment by BenoitCaron** [comment]
Interesting! I didn't think AA would have been the culprit here, since the smaller font is fine. I'll try another build without AA :)

Edit: I did try without AA, and sure enough the text is now very crisp. I still hope this can be fixed someday as the game looks so much nicer (and more faithful to the N64 rendering) with AA enabled :)

**Comment by vaguerant** [comment]
Turning off AA makes the issue less noticeable, but the text is still scaling incorrectly. For a point of comparison, I took a native resolution screenshot in the paraLLEl emulator and overlaid it on the equivalent screen on 3DS:

![Mario64vs3DS](https://user-images.githubusercontent.com/5259025/89647262-a7671700-d900-11ea-8e38-b036f55de243.gif)

As you can see, the baseline of the text (i.e. the lowest pixel on each line) is correct, but each line of text is being scaled to 1 single pixel larger than it was on N64, which is causing these issues.

**EDIT:** Explanation from devwizard on the Discord:
> anyway, the reason is just due to the texture coordinates being scaled slightly smaller than they're supposed to because of a bug in sm64, to fix it youll have to set the texture scale to 0x8000 and then double the texture coordinates for the text
> interestingly, the bug does not exist in the japanese and shindou versions of the game, no clue why they changed it at all

**Comment by BenoitCaron** [comment]
Nice to know we have a theoretical fix! Can anyone try and implement it?
